vendor/README.md

```diff
@@ -0,0 +1,353 @@
+  var userRepo = match[2];  // eg, espadrine/sc
+  var branch = match[3];
+  var format = match[4];
+  var options = {
+    method: 'HEAD',
+    uri: 'https://api.travis-ci.org/' + userRepo + '.svg',
+  };
+  if (branch != null) {
+    options.uri += '?branch=' + branch;                         // (3)
+  }
+  var badgeData = getBadgeData('build', data);
+  request(options, function(err, res) {
+    if (err != null) {
+      console.error('Travis error: ' + err.stack);              // 5
+      if (res) { console.error(''+res); }
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var state = res.headers['content-disposition']
+                     .match(/filename="(.+)\.svg"/)[1];
+      badgeData.text[1] = state;
+      if (state === 'passing') {
+        badgeData.colorscheme = 'brightgreen';                  // 1
+      } else if (state === 'failing') {
+        badgeData.colorscheme = 'red';                          // 1
+      } else {
+        badgeData.text[1] = state;                              // 1, 2, 3
+      }
+      sendBadge(format, badgeData);
+
+    } catch(e) {
+      badgeData.text[1] = 'invalid';                            // 4
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+```
+
+It handles the typical cases (1), resource not found (2), customization (3),
+malformed responses (4), and connection errors (5).
+
+Before getting started, install the project dependencies if you haven't
+already:
+
+```
+npm i
+```
+
+We'll start by creating a new module, `vendor/travis.js`, using this
+boilerplate:
+
+```js
+'use strict';
+
+const Joi = require('joi');                                        // 1
+const ServiceTester = require('./runner/service-tester');          // 2
+
+const t = new ServiceTester({ id: 'travis', title: 'Travis CI' })  // 3
+module.exports = t;                                                // 4
+```
+
+We'll import [Joi][] (1) which will help with our assertions. We'll add all
+our tests to this ServiceTester object (2), which gets exported from the
+module (4). The first attribute passed to the constructor (3) is the id of
+a service, which is used to identify it on the command line or in a pull
+request. The tester will prepend the id to the URIs you provide later, which
+saves copying and pasting. The second attribute is the human-readable title
+of the service, which prints when you run the tests.
+
+Next we'll add a test for the typical case.
+
+[![](https://img.shields.io/badge/build-passing-brightgreen.svg)]()
+
+The JSON format for this badge is `{ name: 'build', value: 'passing' }`.
+
+Here's what our first test looks like:
+
+```js
+t.create('build status on default branch')
+  .get('/rust-lang/rust.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('build'),
+    value: Joi.equal('failing', 'passing', 'unknown')
+  }));
+```
+
+We're using `rust` for our test, though we could have chosen any stable
```

Comment:
We could use shields, though I chose something that was far removed from the project to reduce cognitive load and eliminate the possibility of a beginner being confused.